packages/table/src/regions.ts

```diff
@@ -548,11 +541,52 @@ export class Regions {
         }
     }
 
+    private static overlapsRegionInternal(regions: IRegion[], query: IRegion, allowPartialOverlap = true) {
```

Comment:
There's already a function called `overlapsRegion`, and I couldn't think of a better name. Hence my tacking `internal` on the end.